framework/test/unit/modules/poa/commands/update_authority.spec.ts

```diff
@@ -0,0 +1,419 @@
+			context = testing
+				.createTransactionContext({
+					stateStore,
+					transaction: buildTransaction({
+						params: buildUpdateAuthorityValidatorParams({
+							newValidators: Array.from(Array(MAX_NUM_VALIDATORS + 1).keys()).map(_ => ({
+								address: utils.getRandomBytes(20),
+								weight: BigInt(1),
+							})),
+						}),
+					}),
+					chainID,
+				})
+				.createCommandVerifyContext<UpdateAuthorityParams>(updateAuthoritySchema);
+
+			await expect(updateAuthorityCommand.verify(context)).rejects.toThrow(
+				`newValidators length must be between 1 and ${MAX_NUM_VALIDATORS} (inclusive)`,
+			);
+		});
+
+		it('should throw error when newValidators are not lexicographically ordered', async () => {
+			context = testing
+				.createTransactionContext({
+					stateStore,
+					transaction: buildTransaction({
+						params: buildUpdateAuthorityValidatorParams({
+							newValidators: [
+								{
+									address: address1,
+									weight: BigInt(1),
+								},
+								{
+									address: address0,
+									weight: BigInt(1),
+								},
+							],
+						}),
+					}),
+					chainID,
+				})
+				.createCommandVerifyContext<UpdateAuthorityParams>(updateAuthoritySchema);
+
+			await expect(updateAuthorityCommand.verify(context)).rejects.toThrow(
+				'Addresses in newValidators are not lexicographically ordered.',
+			);
+		});
+
+		it('should throw error when addresses are in newValidators are not unique', async () => {
+			context = testing
+				.createTransactionContext({
+					stateStore,
+					transaction: buildTransaction({
+						params: buildUpdateAuthorityValidatorParams({
+							newValidators: [
+								{
+									address: address0,
+									weight: BigInt(1),
+								},
+								{
+									address: address1,
+									weight: BigInt(1),
+								},
+								{
+									address: address1,
+									weight: BigInt(1),
+								},
+							],
+						}),
+					}),
+					chainID,
+				})
+				.createCommandVerifyContext<UpdateAuthorityParams>(updateAuthoritySchema);
+
+			await expect(updateAuthorityCommand.verify(context)).rejects.toThrow(
+				'Addresses in newValidators are not unique.',
+			);
+		});
+
+		it('should throw error when validator is not in ValidatorStore', async () => {
+			context = testing
+				.createTransactionContext({
+					stateStore,
+					transaction: buildTransaction({
+						params: buildUpdateAuthorityValidatorParams({
+							newValidators: [
+								...updateAuthorityValidatorParams.newValidators,
+								{
+									address: address2,
+									weight: BigInt(2),
+								},
+							],
+						}),
+					}),
+					chainID,
+				})
+				.createCommandVerifyContext<UpdateAuthorityParams>(updateAuthoritySchema);
+
+			await expect(updateAuthorityCommand.verify(context)).rejects.toThrow(
+				`${address.getLisk32AddressFromPublicKey(address2)} does not exist in validator store.`,
+			);
+		});
+
+		it('should throw error when totalWeight is greater than MAX_UINT64', async () => {
```

Comment:
As I commented before, error is mainly read by users, while this unit test caption is read by developers. So to me `totalWeight` is appropriate here so that developers can quickly figure out it's referring to a param